fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexMaintainer.java

```diff
@@ -96,6 +96,20 @@ public Subspace getSecondarySubspace() {
                                                   @Nullable byte[] continuation,
                                                   @Nonnull ScanProperties scanProperties);
 
+    /**
+     * Scan entries in the index.
+     * @param scanBounds the {@link IndexScanBounds bounds} of the scan to perform
+     * @param continuation any continuation from a previous scan invocation
+     * @param scanProperties skip, limit and other properties of the scan
+     * @return a cursor over index entries in the given range
+     */
+    @Nonnull
+    public RecordCursor<IndexEntry> scan(@Nonnull IndexScanBounds scanBounds,
+                                         @Nullable byte[] continuation,
+                                         @Nonnull ScanProperties scanProperties) {
+        return scan(scanBounds.getScanType(), ((IndexScanRange)scanBounds).getScanRange(), continuation, scanProperties);
```

Comment:
Maybe a `ClassCastException` is fine, but it feels like there should be a more descriptive error if a `scanBounds` is given which is not an `IndexScanRange`. At the very least, getting the index name as one of the exception keys is probably a good idea.